jbpm-wb-process-runtime/jbpm-wb-process-runtime-backend/src/main/java/org/jbpm/workbench/pr/backend/server/ProcessDefinitionDataSetProvider.java

```diff
@@ -0,0 +1,187 @@
+
+    @Inject
+    protected ProcessRuntimeDataService processRuntimeDataService;
+
+    @Override
+    public DataSetProviderType getType() {
+        return new ProcessDefinitionDataSetProviderType();
+    }
+
+    @Override
+    public DataSetMetadata getDataSetMetadata(DataSetDef def) throws Exception {
+
+        List<String> columnNames = new ArrayList<>();
+        List<ColumnType> columnTypes = new ArrayList<>();
+
+        List<DataColumnDef> columns = def.getColumns();
+        for (DataColumnDef column : columns) {
+            columnNames.add(column.getId());
+            columnTypes.add(column.getColumnType());
+        }
+
+        return new DataSetMetadataImpl(def,
+                                       def.getUUID(),
+                                       -1,
+                                       def.getColumns().size(),
+                                       columnNames,
+                                       columnTypes,
+                                       -1);
+    }
+
+    @Override
+    public DataSet lookupDataSet(DataSetDef def, DataSetLookup lookup) throws Exception {
+
+        ProcessDefinitionDataSetDef processDefinitionDataSetDef = (ProcessDefinitionDataSetDef) def;
+        final DataSetBuilder dsBuilder = DataSetFactory.newDataSetBuilder();
+        for (final DataColumnDef columnDef : processDefinitionDataSetDef.getColumns()) {
+            dsBuilder.column(columnDef.getId(),
+                             columnDef.getColumnType());
+        }
+
+        ConsoleDataSetLookup consoleDataSetLookup = (ConsoleDataSetLookup)lookup;
+        processDefinitionDataSetDef.setServerTemplateId(consoleDataSetLookup.getServerTemplateId());
+        if (consoleDataSetLookup.getServerTemplateId() == null || consoleDataSetLookup.getServerTemplateId().isEmpty()) {
+            return buildDataSet(def,
+                                new ArrayList<>(),
+                                new ArrayList<>());
+        }
+        String defaultColumnSort = COL_ID_PROCESSNAME;
+        boolean defaultSortOrder = false;
+
+        DataSetSort dataSetSort = lookup.getFirstSortOp();
+        if (dataSetSort != null && dataSetSort.getColumnSortList().size() > 0) {
+            defaultColumnSort = dataSetSort.getColumnSortList().get(0).toString();
+            defaultSortOrder = dataSetSort.getColumnSortList().get(0).getOrder().equals(SortOrder.ASCENDING);
+        }
+
+        final StringBuffer searchText = new StringBuffer();
+        for (DataSetFilter filter : lookup.getOperationList(DataSetFilter.class)) {
+            filter.getColumnFilterList().forEach(columnFilter -> {
+                if (columnFilter instanceof CoreFunctionFilter) {
+                    CoreFunctionFilter coreFunctionFilter = (CoreFunctionFilter) columnFilter;
+                    if (coreFunctionFilter.getParameters().size() > 0) {
+                        searchText.append(coreFunctionFilter.getParameters().get(0));
```

Comment:
I think there will be only one serach String at time to filter, should we just get that from the parameters instead of using a StringBuffer and append?